libsignal-protocol/tests/libsignal-protocol-c-tests.rs

```diff
@@ -400,3 +411,124 @@ fn test_optional_one_time_pre_key() {
     let has_pre_key_id = incoming_message.has_pre_key_id();
     assert!(!has_pre_key_id);
 }
+
+#[test]
+fn test_decrypt() {
```

Comment:
Is this a port of one of the original `libsignal-protocol-c` tests?

Reply:
No it is not. I realize now that the other tests have corresponding tests in the `libsignal-protocol-c`. I think I could rewrite this to match `test_basic_pre_key_v3`. That test seems to match `test_decrypt` quite closely.

Reply:
Na, that's alright. I think most of the other tests were copied from C because that was the most readily accessible source of code which uses this library and shows what it expects to happen.

TL;DR: There's no reason for the correspondence other than Michael being lazy and copying existing tests :P

Reply:
I did rewrite the test and rename it `test_basic_pre_key_v3` in d5bd540b1. It is not a one-to-one mapping as the `libsignal-protocol-c` operates at a lower level, but the basic structure should (roughly) match.

This test also includes serialization and deserialization.